internal/controllers/machine/machine_controller.go

```diff
@@ -91,6 +91,7 @@ type Reconciler struct {
 	// nodeDeletionRetryTimeout determines how long the controller will retry deleting a node
 	// during a single reconciliation.
 	nodeDeletionRetryTimeout time.Duration
+	disableNodeLabelSync     bool
```

Comment:
As a follow-up let's open an issue to track the migration of tests to envclient and the removal of this flag (another option might be to split up the func and then unit test the individual parts vs testing the entire controller).

In the meantime, we can use a follow-up PR to add a comment on this field and make it explicit that it is only for allowing partial testing of the controller logic using the fake client (because features based on SSA cannot be tested with it).

cc @ykakarap

Reply:
Follow-up PR to add the comment: https://github.com/kubernetes-sigs/cluster-api/pull/7965

Issue to migrate tests to envtest and drop the flag: https://github.com/kubernetes-sigs/cluster-api/issues/7964  (also added to the tracking issue).